ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceNotifier.java

```diff
@@ -43,25 +39,22 @@
   private final ExecutionEngineChannel executionEngineChannel;
   private final RecentChainData recentChainData;
   private final PayloadAttributesCalculator payloadAttributesCalculator;
+  private final Spec spec;
 
-  private Optional<ForkChoiceState> forkChoiceState = Optional.empty();
-  private Optional<PayloadAttributes> payloadAttributes = Optional.empty();
-
-  private Optional<ForkChoiceState> lastSentForkChoiceState = Optional.empty();
-  private Optional<PayloadAttributes> lastSentPayloadAttributes = Optional.empty();
-  private SafeFuture<Optional<Bytes8>> lastFuturePayloadId =
-      SafeFuture.completedFuture(Optional.empty());
+  private ForkChoiceUpdateData forkChoiceUpdateData = new ForkChoiceUpdateData();
+  private long payloadAttributesSequenceProducer = 0;
+  private long payloadAttributesSequenceConsumer = -1;
```

Comment:
Wondering if this could/should be done inside `ForkChoiceUpdateData`. Almost certainly a step to far for this PR and possibly not right but maybe worth trying to see how it pans out.

Reply:
I did an implementation of that idea, but it blocks the send, not the potential overwrite of the data itself...
https://github.com/ConsenSys/teku/pull/4775/commits/5a4801583924e6bcfe3806699096bf947cdb7e9f

maybe a similar approach but preventing the attributes to be updated instead...